patterns/disposer.md

```diff
@@ -0,0 +1,84 @@
+# The Promise Disposer Pattern
+
+Often we have to perform resource management in code. For example - release a DB connection:
+
+
+```js
+var users = getDb().then(conn => 
+    conn.query("SELECT name FROM users").then(users => {
+      conn.release();
+      return users;
+    }, e => {
+      conn.release();
+      throw e;
+    })
+});
+```
+
+The problem with the above approach is that if you forget releasing the connection after every single
+time you perform `getDb` you have a resource leak that might freeze your app eventually
+when it runs out of the resource you're leaking.
```

Comment:
I would add that its not just about forgetting, its about easily writing code that stays resilient in the face of potential exceptions, which is an important concern when promises are used. Because we don't crash and we have more exceptions flying around, its far more likely that anything can unsuspectingly interrupt the execution flow at any time.